PR: https://github.com/odoo/odoo/pull/234957

From: 9d035446e9d596aa1990f2682954b4e3c6dc2765
From: Lulu Grimalkin (lugr)
Date: 2026-02-11 07:11:30

Breaking data model changes scores: del:3 + add:0, change matches:
-    consumption = fields.Selection([
-    consumption = fields.Selection([
-    consumption = fields.Selection(related='production_id.consumption')

Total Changes: 40

[IMP] mrp: remove BoM consumption warning field

We're removing the `consumption` field from BoMs, thereby getting rid of
the 'strict' and 'flexible' options. The former 'warning' option is the
default behaviour now. It can still be bypassed through the
`skip_consumption` context parameter.

Task ID: 4441090

Part-of: odoo/odoo#234957
Related: odoo/enterprise#99103
Related: odoo/upgrade#8844
Signed-off-by: Tiffany Chang (tic) <tic@odoo.com>

================================= pseudo patch: =================================

--- a/addons/mrp/models/mrp_bom.py
+++ b/addons/mrp/models/mrp_bom.py
@@ -67,19 +67,6 @@ class MrpBom(models.Model):
     company_id = fields.Many2one(
         'res.company', 'Company', index=True,
         default=lambda self: self.env.company)
-    consumption = fields.Selection([
-        ('flexible', 'Allowed'),
-        ('warning', 'Allowed with warning'),
-        ('strict', 'Blocked')],
-        help="Defines if you can consume more or less components than the quantity defined on the BoM:\n"
-             "  * Allowed: allowed for all manufacturing users.\n"
-             "  * Allowed with warning: allowed for all manufacturing users with summary of consumption differences when closing the manufacturing order.\n"
-             "  Note that in the case of component Highlight Consumption, where consumption is registered manually exclusively, consumption warnings will still be issued when appropriate also.\n"
-             "  * Blocked: only a manager can close a manufacturing order when the BoM consumption is not respected.",
-        default='warning',
-        string='Flexible Consumption',
-        required=True
-    )
     possible_product_template_attribute_value_ids = fields.Many2many(
         'product.template.attribute.value',
         compute='_compute_possible_product_template_attribute_value_ids')

--- a/addons/mrp/models/mrp_production.py
+++ b/addons/mrp/models/mrp_production.py
@@ -258,14 +258,6 @@ class MrpProduction(models.Model):
     production_location_id = fields.Many2one('stock.location', "Production Location", compute="_compute_production_location", store=True)
     picking_ids = fields.Many2many('stock.picking', compute='_compute_picking_ids', string='Picking associated to this manufacturing order')
     delivery_count = fields.Integer(string='Delivery Orders', compute='_compute_picking_ids')
-    consumption = fields.Selection([
-        ('flexible', 'Allowed'),
-        ('warning', 'Allowed with warning'),
-        ('strict', 'Blocked')],
-        required=True,
-        readonly=True,
-        default='flexible',
-    )
 
     mrp_production_child_count = fields.Integer("Number of generated MO", compute='_compute_mrp_production_child_count')
     mrp_production_source_count = fields.Integer("Number of source MO", compute='_compute_mrp_production_source_count')
@@ -1597,8 +1589,6 @@ class MrpProduction(models.Model):
         workorder_ids_to_confirm = set()
         for production in self:
             production_vals = {}
-            if production.bom_id:
-                production_vals.update({'consumption': production.bom_id.consumption})
             # In case of Serial number tracking, force the UoM to the UoM of product
             if production.product_tracking == 'serial' and production.uom_id != production.product_id.uom_id:
                 production_vals.update({
@@ -1755,7 +1745,7 @@ class MrpProduction(models.Model):
         if self.env.context.get('skip_consumption', False):
             return issues
         for order in self:
-            if order.consumption == 'flexible' or not order.bom_id or not order.bom_id.bom_line_ids:
+            if not order.bom_id or not order.bom_id.bom_line_ids:
                 continue
             expected_move_values = order._get_moves_raw_values()
             expected_qty_by_product = defaultdict(float)
@@ -1789,7 +1779,6 @@ class MrpProduction(models.Model):
             lines.append((0, 0, {
                 'mrp_production_id': order.id,
                 'product_id': product_id.id,
-                'consumption': order.consumption,
                 'uom_id': product_id.uom_id.id,
                 'product_consumed_qty_uom': consumed_qty,
                 'product_expected_qty_uom': expected_qty
@@ -1866,19 +1855,6 @@ class MrpProduction(models.Model):
                 filtered_documents[(parent, responsible)] = rendering_context
             production._log_manufacture_exception(filtered_documents, cancel=True)
 
-        # In case of a flexible BOM, we don't know from the state of the moves if the MO should
-        # remain in progress or done. Indeed, if all moves are done/cancel but the quantity produced
-        # is lower than expected, it might mean:
-        # - we have used all components but we still want to produce the quantity expected
-        # - we have used all components and we won't be able to produce the last units
-        #
-        # However, if the user clicks on 'Cancel', it is expected that the MO is either done or
-        # canceled. If the MO is still in progress at this point, it means that the move raws
-        # are either all done or a mix of done / canceled => the MO should be done.
-        mos_to_mark_as_done = self.filtered(lambda p: p.state not in ['done', 'cancel'] and p.bom_id.consumption == 'flexible')
-        if mos_to_mark_as_done:
-            mos_to_mark_as_done.write({'state': 'done'})
-
         return True
 
     def _get_document_iterate_key(self, move_raw_id):

--- a/addons/mrp/models/mrp_workorder.py
+++ b/addons/mrp/models/mrp_workorder.py
@@ -128,7 +128,6 @@ class MrpWorkorder(models.Model):
     production_date = fields.Datetime('Production Date', compute='_compute_production_date', store=True)
     json_popover = fields.Char('Popover Data JSON', compute='_compute_json_popover')
     show_json_popover = fields.Boolean('Show Popover?', compute='_compute_json_popover')
-    consumption = fields.Selection(related='production_id.consumption')
     qty_reported_from_previous_wo = fields.Float('Carried Quantity', digits='Product Unit', copy=False,
         help="The quantity already produced awaiting allocation in the backorders chain.")
     is_planned = fields.Boolean(related='production_id.is_planned')
